### ISSUE TITLE
Incomplete info for dbs block insertion (vocms026)

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#7075]
It's the second time we have this issue with the relval agent. DBS3Upload is failing to update some blocks because they are either missing the acquisition_era or the start_date information.

Small part of the component log
```
2016-08-18 14:28:32,621:140404496717568:ERROR:DBSUploadPoller:Error trying to process block /RelValQCD_Pt_80_120_13_HI/CMSSW_8_1_0_pre8-81X_mcRun2_HeavyIon_v1-v1/GEN-SIM#312e69c2-42fe-11e6-836c-02163e016c
c2 through DBS.
HTTP Error 400: BlockInsert: acquisition_era_name and start_date are required.                                             NULL was received from user input. Please correct your data.
2016-08-18 14:28:32,622:140404496717568:ERROR:DBSUploadPoller:Traceback (most recent call last):
  File "/data/srv/wmagent/v1.0.17.patch1/sw/slc6_amd64_gcc493/cms/wmagent/1.0.17.patch1/lib/python2.7/site-packages/WMComponent/DBS3Buffer/DBSUploadPoller.py", line 116, in uploadWorker
    dbsApi.insertBulkBlock(blockDump = block)
  File "/data/srv/wmagent/v1.0.17.patch1/sw/slc6_amd64_gcc493/cms/dbs3-client/3.3.134/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 317, in insertBulkBlock
    return self.__callServer("bulkblocks", data=blockDump, callmethod='POST' )
  File "/data/srv/wmagent/v1.0.17.patch1/sw/slc6_amd64_gcc493/cms/dbs3-client/3.3.134/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 194, in __callServer
    self.__parseForException(http_error)
  File "/data/srv/wmagent/v1.0.17.patch1/sw/slc6_amd64_gcc493/cms/dbs3-client/3.3.134/lib/python2.7/site-packages/dbs/apis/dbsClient.py", line 221, in __parseForException
    raise HTTPError(http_error.url, data['exception'], data['message'], http_error.header, http_error.body)
HTTPError: HTTP Error 400: BlockInsert: acquisition_era_name and start_date are required.                                             NULL was received from user input. Please correct your data.
```

**Comment by amaltaro** [comment]
Two more important details. There is no acquisition_era in the dbsbuffer_dataset table:
```
SQL> select ACQUISITION_ERA from dbsbuffer_dataset where path='/RelValQCD_Pt_80_120_13_HI/CMSSW_8_1_0_pre8-81X_mcRun2_HeavyIon_v1-v1/GEN-SIM';

ACQUISITION_ERA
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------


```
and second, it seems AcquisitionEra was provided only at task level and not at main level for the request nwickram_RVCMSSW_8_1_0_pre8QCD_Pt_80_120_13_HI_160705_141931_2177.
And during assignment, it was the other way around, only provided at main level.
Still to be debugged

**Comment by ticoann** [comment]
FMI.  if the job report has missing acquisition era this could happen. Need to check whether the acquisition era is set in job report.

https://github.com/dmwm/WMCore/blob/master/src/python/WMComponent/JobAccountant/AccountantWorker.py#L367

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/Steps/Executors/CMSSW.py#L271

It seems all the acquisition era is set correctly in 

https://cmsweb.cern.ch/reqmgr2/config?name=nwickram_RVCMSSW_8_1_0_pre8QCD_Pt_80_120_13_HI_160705_141931_2177

except nwickram_RVCMSSW_8_1_0_pre8QCD_Pt_80_120_13_HI_160705_141931_2177.request.schema.AcquisitionEra = 'FAKE'

which I don't think matters though

**Comment by amaltaro** [comment]
The problem is that only the production/processing tasks have an AcqEra properly set, see
```
>>> for t in task.taskIterator():
...   acqera = t.getAcquisitionEra()
...   print("task name %s and acqera %s" % (t.name(), acqera))   
... 
task name QCD_Pt_80_120_13_HI and acqera CMSSW_8_1_0_pre8
task name LogCollectForQCD_Pt_80_120_13_HI and acqera None
task name QCD_Pt_80_120_13_HIMergeFEVTDEBUGoutput and acqera None
task name QCD_Pt_80_120_13_HIFEVTDEBUGoutputMergeLogCollect and acqera None
task name DIGIHI and acqera CMSSW_8_1_0_pre8
task name LogCollectForDIGIHI and acqera None
task name DIGIHIMergeFEVTDEBUGHLToutput and acqera None
task name DIGIHIFEVTDEBUGHLToutputMergeLogCollect and acqera None
task name RECOHI and acqera CMSSW_8_1_0_pre8
task name LogCollectForRECOHI and acqera None
task name RECOHIMergeRECOSIMoutput and acqera None
task name RECOHIRECOSIMoutputMergeLogCollect and acqera None
task name RECOHICleanupUnmergedRECOSIMoutput and acqera None
task name RECOHIMergeDQMoutput and acqera None
task name RECOHIDQMoutputMergeLogCollect and acqera None
task name RECOHIMergeDQMoutputEndOfRunDQMHarvestMerged and acqera None
task name RECOHIMergeDQMoutputMergedEndOfRunDQMHarvestLogCollect and acqera None
task name RECOHICleanupUnmergedDQMoutput and acqera None
task name RECOHIMergeMINIAODSIMoutput and acqera None
task name RECOHIMINIAODSIMoutputMergeLogCollect and acqera None
task name RECOHICleanupUnmergedMINIAODSIMoutput and acqera None
task name DIGIHICleanupUnmergedFEVTDEBUGHLToutput and acqera None
task name QCD_Pt_80_120_13_HICleanupUnmergedFEVTDEBUGoutput and acqera None
```

I don't know how, since we call exactly the same `setAcquisitionEra()` method as in reqmgr1 assignment.
I actually managed to reproduce it once, but then trying again with the same setup (creating with inner AcquisitionEra only and assigning with main AcquisitionEra only) it works just fine...

**Comment by amaltaro** [comment]
I guess I found the issue here. That workflow was possibly assigned without AcqEra (which would explain why the lower level tasks were not set), the validation function doesn't seem to be catching these cases in ReqMgr2. Testing if my fix catches it now...

**Comment by amaltaro** [comment]
Should have been fixed by #7120. Closing
